### ISSUE TITLE
apachectl vs systemctl problems?

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#5555]
Does the problem described at https://github.com/certbot/certbot/issues/5486#issuecomment-363970559 affect the Apache plugin?

**Comment by trinopoty** [comment]
On Ubuntu 16.04, `systemctl` uses `apachectl` to manage the apache daemon. It directly calls (more or less) `apachectl start` and `apachectl stop` to start/stop the apache daemon.
apachectl manages the apache daemon by invoking the apache2 binary located at `/usr/sbin/apache2`.
So, it doesn't really matter which is used to start/stop apache at least on Ubuntu.

**Comment by bmw** [comment]
I actually think we have the same problem here unfortunately at least on Ubuntu 16.04 due to `ExecStop` not being run if the service wasn't started with `ExecStart` ([source](https://www.freedesktop.org/software/systemd/man/systemd.service.html)). To see this, run:

```
sudo apache2ctl -k start && sudo service apache2 stop && pgrep apache
```
The same thing happens if you use `systemctl` directly rather than `service`.

**Comment by trinopoty** [comment]
On further investigation, it seems that systemd is unable to detect apache started with `apachectl`. But I've always mixed the two commands and never encountered any problem.

**Comment by HorlogeSkynet** [comment]
Hello there,

I think there is also a problem with multiple certificates renewal, because when we overload the `pre` and `post` hooks with `systemctl` commands, Apache looks too slow to finish its warming up before getting stopped :

```
2018-03-26 05:21:45,309:DEBUG:certbot.cli:Var pre_hook=systemctl stop apache2.service (set by user).
2018-03-26 05:21:46,744:DEBUG:certbot.cli:Var post_hook=systemctl start apache2.service (set by user).
```

Thus, I don't really know in which confi

**Comment by bmw** [comment]
@HorlogeSkynet, yes, this is a known issue unfortunately if you use standalone with the Apache plugin to obtain multiple certificates. See #2419.

If you were using standalone to workaround TLS-SNI deprecation a couple months ago, you can switch back to just using Apache if you have Certbot 0.21.0 or newer.

**Comment by HorlogeSkynet** [comment]
@bmw thanks for your fast answer, I wasn't up to speed about that... Sorry

--> You are totally right, Certbot has been upgraded recently, so I'll definitely switch back to the Apache authenticator ASAP.

Thanks, have a good evening :ok_hand:

**Comment by SchoolGuy** [comment]
This is an issue for openSUSE-based distros as they don't use apache2ctl but instead use a custom script tailored towards sysconfig and systemd-specific quirks. I would love if certbot could utilize systemd to manage the apache specific bits.
